src/components/ResourceList/components/Item/Item.tsx

```diff
@@ -419,6 +435,41 @@ function isSelected(id: string, selectedItems?: SelectedItems) {
   );
 }
 
+class KeyWatcher {
```

Comment:
I believe I can replace this whole thing with `EventListener` https://github.com/Shopify/polaris-react/blob/master/src/components/EventListener/EventListener.tsx

Reply:
Yeah, that would be great.